app/controllers/course/enrol_requests_controller.rb

```diff
@@ -24,6 +24,7 @@ def create
   # Allow users to withdraw their requests to register for a course that are pending
   # approval/rejection.
   def destroy
+    @enrol_request.is_validation_for_destroy_required = true
```

Comment:
Since course deletion will bypass controller and go straight to model, why not move the `workflow_state == 'pending'` check to the controller instead?

Then we can save on an additional `is_validation_for_destroy_required` attribute which is only set in the controller destroy anyway.

Reply:
as discussed, we decided to move the validation logic for deletion to the controller instead of model, since the validation will only be called while destroying object through controller